src/catchup/CatchupManager.h

```diff
@@ -37,49 +36,15 @@ enum class VerifyLedgerMode
 class CatchupManager
 {
   public:
-    // The two supported styles of catchup. CATCHUP_COMPLETE will replay all
-    // history blocks, from the last closed ledger to the present, when catching
-    // up; CATCHUP_MINIMAL will attempt to "fast forward" to the next BucketList
-    // checkpoint, skipping the history log that happened between the last
-    // closed ledger and the catchup point. This is set by config, default is
-    // CATCHUP_MINIMAL but it should be CATCHUP_COMPLETE for any server with
-    // API clients. See LedgerManager::startCatchUp and its callers for uses.
-    //
-    // CATCHUP_RECENT is a hybrid mode that does a CATCHUP_MINIMAL to a point
-    // in the recent past, then runs CATCHUP_COMPLETE from there forward to
-    // the present.
-    //
-    // CATCHUP_COMPLETE_IMMEDIATE is a version of CATCHUP_COMPLETE that does not
-    // wait for next ledger checkpoint - it uses last checkpoint available in
-    // history. It is not usable in normal mode of stellar-core, but can be used
-    // to download and apply history without joining to SCP. It can be called by
-    // executing `stellar-core --catchup-complete`.
-    enum CatchupMode
-    {
-        CATCHUP_COMPLETE,
-        CATCHUP_COMPLETE_IMMEDIATE,
-        CATCHUP_MINIMAL,
-        CATCHUP_RECENT
-    };
-
     static std::unique_ptr<CatchupManager> create(Application& app);
 
     // Callback from catchup, indicates that any catchup work is done.
     virtual void historyCaughtup() = 0;
 
-    // Run catchup, we've just heard `initLedger` from the network. Mode can be
-    // CATCHUP_COMPLETE, meaning replay history from last to present, or
-    // CATCHUP_MINIMAL, meaning snap to the next state possible and discard
-    // history. See larger comment above for more detail.
-    //
-    // The `manualCatchup` flag modifies catchup behavior to avoid rounding up
-    // to the next scheduled checkpoint boundary, instead catching up to a
-    // checkpoint presumed to have been made at `initLedger` (i.e. with
-    // checkpoint ledger number equal to initLedger-1). This 'manual' catchup
-    // mode exists to support catching-up to manually created checkpoints.
-    virtual void catchupHistory(uint32_t initLedger, CatchupMode mode,
-                                CatchupWork::ProgressHandler handler,
-                                bool manualCatchup = false) = 0;
+    // Run catchup with given configuration and verify mode.
+    virtual void catchupHistory(CatchupConfiguration catchupConfiguration,
+                                VerifyLedgerMode verifyMode,
```

Comment:
any reason to not incorporate verifyMode with CatchupConfiguration, I don't see why we keep it as a separate field